airflow/www/views.py

```diff
@@ -3574,13 +3600,32 @@ def datasets_summary(self):
                     DatasetModel.id,
                     DatasetModel.uri,
                 )
-                .filter(DatasetModel.uri.ilike(f"%{uri_pattern}%"))
                 .order_by(*order_by)
-                .offset(offset)
-                .limit(limit)
-                .all()
-            ]
-            data = {"datasets": datasets, "total_entries": total_entries}
+            )
+
+            if uri_pattern:
+                count_query = count_query.filter(DatasetModel.uri.ilike(f"%{uri_pattern}%"))
+                query = query.filter(DatasetModel.uri.ilike(f"%{uri_pattern}%"))
+
+            if updated_after:
+                count_query = count_query.outerjoin(
```

Comment:
adding these filters presents an ambiguity.

do you still return all the datasets, but the filters only apply to the counts?
or, do you just return the datasets with updates in the time range.

personally i think the more intuitive behavior is to return only those datasets with updates in the period.  but, i guess this is just an internal API so really we have to look at why we're adding this and how it's gonna be used.

so in short, are you sure you want a left join here?

Reply:
er, actually, now that i think of it, if the user is providing before or after, then it will in effect be an inner join anyway, so may as well make it an inner join here.

Reply:
and... if you don't want that behavior, then you'd want to move the filter from where clause to join condition

Reply:
Good call - yep I think we want an inner join in there.